extension.ts

```diff
@@ -0,0 +1,303 @@
+      rootModule.module.packageLocation,
+      SERVERLESS,
+      FUNCTIONS,
+      route + HTTP_YAML
+    );
+
+    const routeJsonPath = path.join(
+      rootModule.module.packageLocation,
+      BUILD,
+      SERVERLESS,
+      FUNCTIONS,
+      route + HTTP_JSON
+    );
+
+    const doc = await readYamlFileStore(routePath);
+
+    /**
+     * TODO: merge prebuild and build
+     */
+    await writeFile(routeJsonPath, JSON.stringify(doc));
+  }
+};
+
+export const prepare: Hook = async (context: IContext) => {
+  console.log("prepare called");
+  const root = context.moduleHandler.getModule(
+    context.moduleHandler.roodModuleName
+  );
+
+  /**
+   * TODO:parse template.yaml and find functions
+   * then find module names and go to modules and find getconfigs
+   * then generate schema compiled files and save inside .somod
+   * create the layer in template.yaml and give contentUri path of previous step.
+   */
+  const templateFile = path.join(root.module.packageLocation, TEMPLATE_YAML);
+  const template: JSONObjectType = await readYamlFileStore(templateFile);
+
+  const functions = parseFunctions(template);
+
+  await Promise.all(
+    functions.map(async (func: JSONObjectType) => {
+      const props = func[PROPERTIES] as JSONObjectType;
+      let codeUri = props[CODE_URI] as string;
+      codeUri = codeUri.replace(
+        path.join(SOMOD, SERVERLESS, FUNCTIONS) + "/",
+        ""
+      );
+
+      // get module name
+      /**
+       * TODO: recheck this syntax
+       */
+      const ps = codeUri.split("/");
+      ps.pop();
+      const moduleName = ps.join("/");
+
+      //get module path from name
+      const module = context.moduleHandler.getModule(moduleName);
+      const __routeFileNames = await getRouteFiles(module);
+
+      for (const __routeFileName of __routeFileNames) {
+        const _routeFilePath = path.join(
+          module.module.packageLocation,
+          BUILD,
+          SERVERLESS,
+          FUNCTIONS,
+          __routeFileName + HTTP_JSON
+        );
+
+        const _tRoutes = {} as RoutesTransformed;
+        const routes = (await readJsonFileStore(_routeFilePath)) as Routes;
+        await Promise.all(
+          Object.keys(routes).map(async route => {
+            const _rOptions = routes[route] as HttpMethodOptions;
+
+            const _schemaDir = path.join(
+              root.module.packageLocation,
+              SOMOD,
+              SERVERLESS,
+              FUNCTION_LAYERS,
+              moduleName,
+              __routeFileName
+            );
+
+            await Promise.all(
+              Object.keys(_rOptions).map(async method => {
+                const _routekey = `${method} ${route}`;
+                _tRoutes[_routekey] = {};
+
+                await mkdir(_schemaDir, { recursive: true });
+                const _mOptions = _rOptions[method] as KeyOptions;
+
+                await Promise.all(
+                  Object.keys(_mOptions).map(async key => {
```

Comment:
instead of key and generic variable names use `function`, `route`, `method`, `layer`

Reply:
mentioned names are also used and not available while schema parsing